src/service.ts

```diff
@@ -623,11 +627,28 @@ export class DebuggerService implements IDebugger, IDisposable {
   private _modelChanged = new Signal<IDebugger, IDebugger.IModel>(this);
   private _eventMessage = new Signal<IDebugger, IDebugger.ISession.Event>(this);
 
+  private _specsManager: KernelSpec.IManager;
+
   private _hashMethod: (code: string) => string;
   private _tmpFilePrefix: string;
   private _tmpFileSuffix: string;
 }
 
+/**
+ * A namespace for `DebuggerService` statics.
+ */
+export namespace DebuggerService {
+  /**
+   * Instantiation options for a `DebuggerService`.
+   */
+  export interface IOptions {
+    /**
+     * The kernel specs manager.
+     */
+    specsManager: KernelSpec.IManager;
```

Comment:
We might want to make this optional, so instantiating a `DebuggerService` without a kernel specs manager would always assume debugging is enabled (for example in an cutom JupyterFrontEnd where only one kernel is available).